### PR TITLE
Changing build tool to sbt

### PR DESCRIPTION
**Pull request by maxhutch**
Getting test working is trivial.  Most of the complexity is in the adoption of https://github.com/olafurpg/sbt-ci-release for managing the maven central releases.

### CODE REVIEW

**Comment by maxhutch** [comment]
I can't test that the deploys work until this hits master since I configured the secrets to be only available on the master branch.

File: project/build.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sbt.version=1.1.6
```
Comment (andyczerwonka) [inline project/build.properties:1]: Why would we choose such an old version? The latest release is `1.4.5`

Comment (maxhutch) [reply project/build.properties]: I'll bump it to what we use in mithril.

Comment (andyczerwonka) [reply project/build.properties]: They don't need to be in sync. This works, so does the Mithril version or the latest version... was just curious as to why an old version.

**Review by andyczerwonka — APPROVED** [review]

**Pull request merged**